ui/src/cohortSummary.tsx

```diff
@@ -0,0 +1,65 @@
+import Typography from "@mui/material/Typography";
+import { GridBox } from "layout/gridBox";
+import GridLayout from "layout/gridLayout";
+import * as tanagra from "tanagra-api";
+import {
+  getCriteriaPlugin,
+  getCriteriaTitle,
+  getCriteriaTitleFull,
+} from "./cohort";
+
+export type CohortSummaryProps = {
+  cohort: tanagra.Cohort;
+};
+
+export function CohortSummary(props: CohortSummaryProps) {
+  return (
+    <GridLayout rows height="auto">
+      {props.cohort.groupSections.map((s, si) => (
+        <GridLayout key={s.id} rows height="auto">
+          <Typography
+            variant="body2em"
+            component="p"
+            sx={{ "::first-letter": { textTransform: "capitalize" } }}
+          >
+            {si != 0 ? "and " : ""}
+            {s.filter.excluded ? "exclude " : "include "}participants with
+            {s.filter.kind === tanagra.GroupSectionFilterKindEnum.Any
+              ? " any "
+              : " all "}
+            of the following criteria:
+          </Typography>
+          {s.groups.map((g, gi) => (
+            <GridBox key={g.id} sx={{ height: "auto" }}>
+              <Typography variant="body2" component="span">
+                {getCriteriaTitleFull(g.criteria[0])}
+              </Typography>
+              {g.criteria.length > 1 ? (
+                <span>
+                  <Typography variant="body2em" component="span">
+                    {" where "}
+                  </Typography>
+                  <Typography variant="body2" component="span">
+                    {g.criteria
+                      .slice(1)
+                      .map((c) =>
+                        getCriteriaTitle(c, getCriteriaPlugin(c, g.entity))
```

Comment:
Does each criteria plugin define its own title, or can they override the default title somehow?

Reply:
The criteria provides part of the title and it's combined with the criteria type.